packages/nectar/src/sagas/publicChannels/publicChannels.selectors.ts

```diff
@@ -46,64 +41,45 @@ const currentCommunityChannelsState = createSelector(
   }
 )
 
-export const publicChannels = createSelector(
-  currentCommunityChannelsState,
-  (state) => {
-    return publicChannelsAdapter.getSelectors().selectAll(state.channels)
-  }
-)
+export const publicChannels = createSelector(currentCommunityChannelsState, state => {
+  return publicChannelsAdapter.getSelectors().selectAll(state.channels)
+})
 
-export const publicChannelsMessages = createSelector(
-  currentCommunityChannelsState,
-  (state) => {
-    return channelMessagesAdapter
-      .getSelectors()
-      .selectAll(state.channelMessages)
-  }
-)
+export const publicChannelsMessages = createSelector(currentCommunityChannelsState, state => {
+  return channelMessagesAdapter.getSelectors().selectAll(state.channelMessages)
+})
 
-export const missingChannelsMessages = createSelector(
-  publicChannelsMessages,
-  (messages) => {
-    return messages
-      .filter((message) => message.type === MessageType.Empty)
-      .map((message) => message.id)
-  }
-)
+export const missingChannelsMessages = createSelector(publicChannelsMessages, messages => {
+  return messages.filter(message => message.type === MessageType.Empty).map(message => message.id)
+})
 
-export const currentChannel = createSelector(
-  currentCommunityChannelsState,
-  (state) => {
-    return state.currentChannel
-  }
-)
+export const currentChannel = createSelector(currentCommunityChannelsState, state => {
+  return state.currentChannel
+})
 
-export const channelLoadingSlice = createSelector(
-  currentCommunityChannelsState,
-  (state) => {
-    return state.channelLoadingSlice
-  }
-)
+export const channelLoadingSlice = createSelector(currentCommunityChannelsState, state => {
+  return state.channelLoadingSlice
+})
 
 export const currentChannelMessages = createSelector(
   publicChannelsMessages,
   currentChannel,
   (messages, channel) => {
-    return messages.filter((message) => message.channelId === channel)
+    return messages.filter(message => message.channelId === channel)
   }
 )
 
 const validCurrentChannelMessages = createSelector(
   currentChannelMessages,
   certificatesMapping,
   (messages, certificates) => {
-    return messages.filter((message) => message.pubKey in certificates)
+    return messages.filter(message => message.pubKey in certificates)
```

Comment:
One of our security assumptions is that we separate the encryption layer (nectar) from the networking layer (waggle). So that makes nectar the very first place in which you can decrypt and validate the signatures.

Signatures are unique for each message, they're encrypted with the use of the message content data and the user's privateKey. PublicKey that decorates the message must match the signature. Otherwise it won't be considered valid.

I see we have the `verifySignature` method in @zbayapp/identity but it looks like it's not being used anywhere - so that's probably the thing we should take care of

Reply:
created issue. https://github.com/ZbayApp/monorepo/issues/135